maestro-studio/web/src/components/commands/ReplView.tsx

```diff
@@ -10,9 +10,32 @@ import { useDeviceContext } from "../../context/DeviceContext";
 import { useRepl } from '../../context/ReplContext';
 
 const getFlowText = (selected: ReplCommand[]): string => {
-  return selected
-    .map((c) => (c.yaml.endsWith("\n") ? c.yaml : `${c.yaml}\n`))
-    .join("");
+  let combinedYaml = '';
+  
+  // Process each command independently
+  selected.forEach(command => {
+    // Get the first line of the command YAML
+    const lines = command.yaml.split('\n');
+    if (lines.length === 0) return;
+    
+    // Process the first line (command line)
+    let firstLine = lines[0].trim();
+    if (!firstLine.startsWith('-')) {
+      firstLine = `- ${firstLine}`;
+    }
+    combinedYaml += firstLine + '\n';
+    
+    // Process the parameter lines
+    for (let i = 1; i < lines.length; i++) {
+      const line = lines[i].trim();
+      if (line) {
+        // Parameters should have 4-space indentation
+        combinedYaml += '    ' + line + '\n';
```

Comment:
This only works when it's a generated command.

If I input something more complex and nested, this will break the previously good indentation.

Silly example from the Android homescreen:
<img width="351" height="428" alt="image" src="https://github.com/user-attachments/assets/9f11dc4b-e899-4460-adcb-d344c4653c06" />

This:
```
assertVisible:
  text: Gmail
  leftOf:
    text: Photos
    leftOf: YouTube
```
is copied as
```
- assertVisible:
    text: Gmail
    leftOf:
    text: Photos
    leftOf: YouTube
```